src/any_agent/tools/a2a_tool.py

```diff
@@ -0,0 +1,41 @@
+# adapted from https://github.com/google/a2a-python/blob/main/examples/helloworld/test_client.py
+import httpx
+from typing import Any, Callable
+from common.client import A2AClient
+from uuid import uuid4
+
+def a2a_query(url: str, description: str) -> Callable[[str], str]:
```

Comment:
We should not require `description` as arg. There are method to retrieve the agent card from the URL where we could get the info we need to create a descriptive docstring

Reply:
ok, let's retrieve the agent card and get the description from there 👍